src/modules/organisation/domain/organisation.service.ts

```diff
@@ -17,4 +17,12 @@ export class OrganisationService {
         }
         return { ok: false, error: new EntityCouldNotBeCreated(`Organization could not be created`) };
     }
+
+    public async findOrganisationById(id: string): Promise<Result<OrganisationDo<true>, DomainError>> {
+        const organisation: Option<OrganisationDo<true>> = await this.organisationRepo.findById(id);
+        if (organisation) {
+            return { ok: true, value: organisation };
+        }
+        return { ok: false, error: new EntityNotFoundError(`Organization with the following ID ${id} does not exist`) };
```

Comment:
Maybe we could the message for the entity not found error more generic? So that not every caller has a different message.